routers/web/repo/http.go

```diff
@@ -485,7 +485,7 @@ func serviceRPC(ctx gocontext.Context, h serviceHandler, service string) {
 	}
 
 	var stderr bytes.Buffer
-	cmd := git.NewCommandContextNoGlobals(h.r.Context(), service, "--stateless-rpc", h.dir)
```

Comment:
this looks valid before - blocking to have a deeper dig into code ... cc @zeripath

Reply:
I just don't understand why this ever worked and why this fixes things.

Reply:
> I just don't understand why this ever worked

See my recent comment.



> and why this fixes things.

The `NewCommandContext` function add the globals, one of them is to allow the filter functionality.

https://github.com/go-gitea/gitea/blob/5e5740af69cb0f44ff0c1576a2387388d75bb4c2/modules/git/git.go#L149-lL152

Reply:
so it's not a regression - its a independent fix

Reply:
> so it's not a regression - its a independent fix

Seems like it, I've changed the Title + Comment according to it.